gwswlib/hydx.py

```diff
@@ -28,28 +61,29 @@ def _check_on_unique(self, records, unique_field, remove_double=False):
             )
 
 
-class ConnectionNode:
-    MAAIVELDSCHEMATISERINGCOLL = [
-        {"RES": "Reservoir"},
-        {"KNV": "Gekneveld"},
-        {"VRL": "Verlies"},
-    ]
+class Generic:
+    @classmethod
+    def csvheaders(cls):
+        return [field["csvheader"] for field in cls.FIELDS]
 
-    TYPEKNOOPPUNTCOLL = [
-        {"INS": "Inspectieput"},
-        {"ITP": "Infiltratieput"},
-        {"CMP": "Compartiment"},
-        {"UIT": "Uitlaat"},
-    ]
+    def import_csvline(self, csvline):
```

Comment:
Waarschijnlijk gebruik je het op deze manier:

    connection_node = ConnectionNode()  # Ergens is dit een raar, leeg, niet-echt object.
    connection_node.import_csvline(csvline)  # Kijk, nu kan je hem gebruiken.

Als je van import_csvline een class method maakt die een object teruggeeft kan je het als een soort `__init__()` gebruiken:

    connection_node = ConnectionNode.import_csvline(csvline)

De class methode zou zoiets worden:

    @classmethod
    def import_csvline(cls, csvline):
        instance = cls()
        # je bestaande code, met 'instance' i.p.v. 'self'.
        return instance  # Kijk, hij gedraagt zich als een __init__()

Kijk maar of die stijl je bevalt.